File: Programming-languages-and-laboratory/Python/Exercises/31_compito_java_20190705/31_soluzione_python/cliente.py
from noleggio import Noleggio

class Cliente:

    nome_cognome: str
    noleggi: list

    def __init__(self, nome_cognome: str, noleggi: list = None):
        self.nome_cognome = nome_cognome
        if noleggi is None:
            self.noleggi = []
        else:
            self.noleggi = noleggi 

    def add_noleggio(self, noleggio: Noleggio):
        self.noleggi.append(noleggio)

    def costo_noleggio_maggiore(self) -> float:
        max = 0
        result = 0
        for noleggio in self.noleggi:
            if noleggio.costo_giornaliero > max:
                max = noleggio.costo_giornaliero
                result = noleggio.costo_giornaliero * noleggio.numero_giorni

        return result

    
    def __str__(self):
        return self.nome_cognome + ", " + str(self.costo_noleggio_maggiore()) + ", " + str(len(self.noleggi))

File: II - Linguaggi di programmazione e laboratorio/Java/Exercises/40_compito_java_2020_06_22/Cliente.java

import java.io.*;
import java.util.*;

public class Cliente {

    private String nomeCognome;
    private List<Noleggio> noleggi;

    public Cliente(String nomeCognome) {
        this.nomeCognome = nomeCognome;
        this.noleggi = new LinkedList<Noleggio>();
    }

    public void addNoleggio(String targa, int giorni, double costo) {
        noleggi.add(new Noleggio(targa, giorni, costo));
    }

    public double getTot() {
        double tot = 0;
        for (Noleggio n : noleggi)
            tot = tot + n.getTot();
        return tot;

    }

    public String toString() {
        return nomeCognome + " " + getTot() + " " + noleggi.size() + " " + noleggi;
    }
}